### PR TITLE
Fix space tab indentation

### LINKED ISSUE #8426: Formatter doesn't change spaces to tabs if number of spaces matches indent size
**TypeScript Version:** 

1.8.10

**Code**

```ts

import * as ts from "typescript";

// Note: this uses ts.formatting which is part of the typescript 1.4 package but is not currently 
//       exposed in the public typescript.d.ts. The typings should be exposed in the next release. 
function format(text: string) {
    let options = getDefaultOptions();

    // Parse the source text
    let sourceFile = ts.createSourceFile("file.ts", text, ts.ScriptTarget.Latest, /*setParentPointers*/ true);

    // Get the formatting edits on the input sources
    let edits = (<any>ts).formatting.formatDocument(sourceFile, getRuleProvider(options), options);

    // Apply the edits on the input code
    return applyEdits(text, edits);

    function getRuleProvider(options: ts.FormatCodeOptions) {
        // Share this between multiple formatters using the same options.
        // This represents the bulk of the space the formatter uses.
        let ruleProvider = new (<any>ts).formatting.RulesProvider();
        ruleProvider.ensureUpToDate(options);
        return ruleProvider;
    }

    function applyEdits(text: string, edits: ts.TextChange[]): string {
        // Apply edits in reverse on the existing text
        let result = text;
        for (let i = edits.length - 1; i >= 0; i--) {
            let change = edits[i];
            let head = result.slice(0, change.span.start);
            let tail = result.slice(change.span.start + change.span.length)
            result = head + change.newText + tail;
        }
        return result;
    }

    function getDefaultOptions(): ts.FormatCodeOptions {
        return {
            IndentSize: 4,
            IndentStyle: 1,
            TabSize: 4,
            NewLineCharacter: '\n',
            ConvertTabsToSpaces: false,
            InsertSpaceAfterCommaDelimiter: true,
            InsertSpaceAfterSemicolonInForStatements: true,
            InsertSpaceBeforeAndAfterBinaryOperators: true,
            InsertSpaceAfterKeywordsInControlFlowStatements: true,
            InsertSpaceAfterFunctionKeywordForAnonymousFunctions: false,
            InsertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis: false,
            InsertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets: false,
            InsertSpaceAfterOpeningAndBeforeClosingTemplateStringBraces: false,
            PlaceOpenBraceOnNewLineForFunctions: false,
            PlaceOpenBraceOnNewLineForControlBlocks: false,
        };
    }
}


let code = "var a=function(v:number){\n    return 0+1+2+3;\n}";
let result = format(code);
console.log(result);

```

**Expected behavior:**
The `return` line should be indented with tab.

**Actual behavior:** 
It's actually left untouched because at input it is already indented with 4 spaces and IndentSize is set to 4. This is, however, counter-intuitive, the formatter should fix the indentation to tabs, in my opinion.

The reason this happens is in https://github.com/Microsoft/TypeScript/blob/d960200dac73d077a66b8ac6ef0648895dc3493e/src/services/formatting/formatting.ts#L869
where the token start position (4) is compared to expected indent size, but there is no check whether the indentation is made from spaces.

### PR DESCRIPTION
**Pull request by ziacik**
Fixes #8426 
Pls review and merge if possible. The issue hasn't been labeled yet.

### CODE REVIEW

File: src/services/formatting/formatting.ts [hunk 1]
```diff
@@ -866,13 +866,23 @@ namespace ts.formatting {
             }
             else {
                 let tokenStart = sourceFile.getLineAndCharacterOfPosition(pos);
-                if (indentation !== tokenStart.character) {
-                    let startLinePosition = getStartPositionOfLine(tokenStart.line, sourceFile);
+                let startLinePosition = getStartPositionOfLine(tokenStart.line, sourceFile);
+                if (indentation !== tokenStart.character || indentationIsDifferent(indentationString, startLinePosition)) {
```
Comment (mhegazy) [inline src/services/formatting/formatting.ts]: why not just `indentationString === soruceFile.text.substr(startLinePosition , indentationString.length);`

**Comment by ziacik** [comment]
Well, I guess I did some overengineering here, lol. Looking at the rest of the source code which does work with charcodes a lot maybe I thought it would be better to do it like that and maybe that it would have a better performance than creating a substring. That's just a very blind guess, probably. I can change it to your way by another commit.

**Comment by mhegazy** [comment]
the change looks good to me. if you can get that comparison simplified we should be good to go.

thanks for taking the time to fix this issue.

**Comment by ziacik** [comment]
I have added the simplification commit.

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**